File: sdk-js/examples/summarize-file-url.ts
import { getModel } from "./get-model.ts";

const provider = process.env["PROVIDER"] ?? "openai";
const modelId = process.env["MODEL"] ?? "gpt-5.6-sol";
const model = getModel(provider, modelId);

const response = await model.generate({
  messages: [
    {
      role: "user",
      content: [
        { type: "text", text: "Summarize the attached PDF." },
        {
          type: "file",
          // The provider fetches this URL; no download or base64 encoding is needed here.
          url: "https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf",
          mime_type: "application/pdf",
        },
      ],
    },
  ],
});

console.dir(response, { depth: null });

File: sdk-rust/examples/summarize-file-url.rs
use dotenvy::dotenv;
use llm_sdk::{LanguageModelInput, Message, Part};

mod common;

#[tokio::main]
async fn main() {
    dotenv().ok();

    let provider = std::env::var("PROVIDER").unwrap_or_else(|_| "openai".to_string());
    let model_id = std::env::var("MODEL").unwrap_or_else(|_| "gpt-5.6-sol".to_string());
    let model = common::get_model(&provider, &model_id);

    let response = model
        .generate(LanguageModelInput::new([Message::user([
            Part::text("Summarize the attached PDF."),
            // The provider fetches this URL; no download or base64 encoding is needed here.
            Part::file_from_url(
                "https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf",
                "application/pdf",
            ),
        ])]))
        .await
        .expect("Generation failed");

    println!("{response:#?}");
}
